crates/egui/src/widgets/text_edit/builder.rs

```diff
@@ -1025,6 +1032,40 @@ fn events(
 
 // ----------------------------------------------------------------------------
 
+fn ime_enabled_filter_events(events: Vec<Event>) -> Vec<Event> {
+    let mut filter_events: Vec<Event> = Vec::new();
+
+    for event in events {
+        match event {
+            Event::Key { repeat: true, .. }
+            | Event::Key {
+                key: Key::Backspace,
+                ..
+            } => {}
+            _ => filter_events.push(event),
+        }
+    }
+
+    filter_events
+}
+
+fn ime_front_events(events: Vec<Event>) -> Vec<Event> {
+    let mut ime_front_events: Vec<Event> = Vec::new();
+    let mut other_events: Vec<Event> = Vec::new();
+
+    for event in events {
+        match event {
+            Event::Ime(_) => ime_front_events.push(event),
+            _ => other_events.push(event),
+        }
+    }
+
+    ime_front_events.extend(other_events);
+    ime_front_events
+}
```

Comment:
this could be a `.sort(…)`

Reply:
> this could be a `.sort(…)`

Since there shouldn't be any change in order, it would be better to simply process it without using `sort()`. 
Some modifications have been made.